src/yunohost/app.py

```diff
@@ -87,27 +88,29 @@ def app_fetchlist(url=None, name=None):
 
     """
     # Create app path if not exists
-    try: os.listdir(repo_path)
-    except OSError: os.makedirs(repo_path)
+    if not os.path.exists(repo_path):
+        os.makedirs(repo_path)
 
     if url is None:
         url = 'https://app.yunohost.org/official.json'
         name = 'yunohost'
-    else:
-        if name is None:
-            raise MoulinetteError(errno.EINVAL,
-                                  m18n.n('custom_appslist_name_required'))
+    elif name is None:
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('custom_appslist_name_required'))
 
+    # Download file
+    try:
+        applist = requests.get(url, timeout=30).text
```

Comment:
using `requests` instead of `wget` does not trigger 404 error nor no json file. Try `yunohost app fetchlist -u https://app.yunohost.org/404.json -n 404list` for example